### PR TITLE
Project Discussion Vue

### PR DESCRIPTION
**Pull request by felixoi**

### CODE REVIEW

File: oreClient/src/main/assets/api.js [hunk 1]
```diff
@@ -17,7 +17,10 @@ export class API {
                     contentType: isFormData ? false : 'application/json',
                     data: isBodyRequest && !isFormData ? JSON.stringify(data) : data,
                     processData: !(isFormData || isBodyRequest),
-                    headers: {'Authorization': 'OreApi session=' + session}
+                    headers: {
+                        'Authorization': 'OreApi session=' + session,
+                        'Csrf-Token': csrf
```
Comment (Katrix) [inline oreClient/src/main/assets/api.js:22]: What is this here for?

File: oreClient/src/main/assets/pages/project/ProjectDiscussion.vue [hunk 2]
```diff
@@ -1,78 +1,298 @@
...
+        watch: {
+            project(val, oldVal) {
+                if(!oldVal || val.plugin_id !== oldVal.plugin_id) {
+                    this.updateData(val);
+                }
+            }
+        },
+        created() {
+            if(this.project) {
+                NProgress.start();
```
Comment (Katrix) [inline oreClient/src/main/assets/pages/project/ProjectDiscussion.vue]: Remove this? Also started in `updateData`?

File: oreClient/yarn.lock [hunk 3]
```diff
@@ -2485,6 +2485,11 @@ emoji-regex@^7.0.1:
   resolved "https://registry.yarnpkg.com/emoji-regex/-/emoji-regex-7.0.3.tgz#933a04052860c85e83c122479c4748a8e4c72156"
   integrity sha512-CwBLREIQ7LvYFB0WyRvwhq5N5qPhc6PMjD6bYggFlI5YyDgl+0vxq5VHbMOFqLg7hfWzmu8T5Z1QofhmTIhItA==
 
+emojis-list@^2.0.0:
```
Comment (Katrix) [inline oreClient/yarn.lock:2488]: We've added emoji's to Ore. Do these also work in version and project pages?

**Review by Katrix — CHANGES_REQUESTED** [review]
Just an initial skim

File: oreClient/src/main/assets/api.js (hunk 1 above)
Comment (felixoi) [reply oreClient/src/main/assets/api.js:22]: If you do something which is not GET its needed. Before this was added globally but now I do Ajax requests on discourse which doesn't accept the header.

File: oreClient/yarn.lock (hunk 3 above)
Comment (felixoi) [reply oreClient/yarn.lock:2488]: I didn't do this one. This was yarn itself

File: oreClient/src/main/assets/pages/project/ProjectDiscussion.vue [hunk 4]
```diff
@@ -1,78 +1,297 @@
...
+                }).fail((xhr) => {
+                    // TODO
+                    console.log(xhr)
+                })
+            }
+        },
+        watch: {
+            project(val, oldVal) {
+                if(!oldVal || val.plugin_id !== oldVal.plugin_id) {
+                    this.updateData(val);
```
Comment (Katrix) [inline oreClient/src/main/assets/pages/project/ProjectDiscussion.vue]: You're passing in the whole object here instead of just the topic id

**Pull request merged**